spec/features/ncr/work_orders/create_spec.rb

```diff
@@ -159,6 +159,19 @@
       expect(page).to have_content('$2,000 for construction')
     end
 
+    scenario "selecting Expense Type toggles Building required flag", :js do
```

Comment:
It seems like this one test is testing a few different things. If it were me, I might break out out like this:
```
context "expense type is BA60" do
  scenario "building id is not required" do
     select_expense_type("ba60")
    find("#ncr_work_order_expense_type_ba60").click
    expect_building_id_not_to_be_required
  end
end

context "expense type is not BA60" do
  scenario "building id is required" do
    ["ba61", "ba80"].each do |expense_type|
      select_expense_type(expense_type)
      expect_building_id_to_be_required
    end
  end
end

context "selects BA60 and then unselects BA60" do
  scenario "building id is required" do
    select_expense_type("ba60")
    select_expense_type("ba80")
     expect_building_id_to_be_required
  end
end

def select_expense_type(expense_type)
  find("#ncr_work_order_expense_type_#{expense_type}").click
end

def expect_building_id_not_to_be_required
  expect(find('.ncr_work_order_building_number input')['class']).to_not match('required')
end

def expect_building_id_to_be_required
  expect(find('.ncr_work_order_building_number input')['class']).to match('required')
end
```

what do you think?